static/js/click_handlers.js

```diff
@@ -649,18 +657,34 @@ exports.initialize = function () {
                 }
             }
         });
-
+        function group_or_stream(temp, selector, flag) {
+            var make_or_finish_editable = ["editable", "checkmark"];
+            var group_id = $(temp).closest('.user-group').attr("id");
+            if (group_id) {
+                if (selector === '.group-name-editable') {
+                    return "#user-groups #" + group_id + " .group-name ." + make_or_finish_editable[flag];
+                } else if (selector === '.group-description-editable') {
+                    return "#user-groups #" + group_id + " .group-description ." + make_or_finish_editable[flag];
+                }
+            } else if (!group_id) {
+                if (flag === 1) {
+                    return "[data-finish-editing='" + selector + "']";
+                } else if (flag === 0) {
+                    return "[data-make-editable='" + selector + "']";
+                }
+            }
+        }
```

Comment:
I believe that the reason you need this function is because there are more than one user groups rendered together in one page, where there will be always one stream setting open at any time. So you require `user-group-id` to filter user group element, and there is no need of `stream-id` as only one present in template. 
Is this the reason?

Reply:
yeah :+1: , I have elaborated it in my 3rd commit message too.